imgix/urlbuilder.py

```diff
@@ -52,13 +52,30 @@ def __init__(
         if not isinstance(domains, (list, tuple)):
             domains = [domains]
 
+        self.validate_domain(domains)
         self._domains = domains
         self._sign_key = sign_key
         self._use_https = use_https
         self._shard_strategy = shard_strategy
         self._shard_next_index = 0
         self._sign_with_library_version = sign_with_library_version
 
+    def validate_domain(self, domain):
+        regex = re.compile(
+            r'^(?:[a-z\d\-_]{1,62}\.){0,125}'
+            r'(?:[a-z\d](?:\-(?=\-*[a-z\d])|[a-z]|\d){0,62}\.)'
+            r'[a-z\d]{1,63}$'
+        )
+        err_str = str(
+            'Domain url does not conform to the required structure.\n'
+            'Be sure to confirm that the protocol (http/https) is not '
+            'prepended and/or that part of the image path is not '
+            'included in the url: ')
+
+        for url in domain:
```

Comment:
Let's use `for domain in domains:` here, since this isn't really handling URLs at all.

Reply:
Agreed